solutions/usaco-600.mdx

```diff
@@ -2,19 +2,105 @@
 id: usaco-600
 source: USACO Platinum 2016 January
 title: Fort Moo
-author: Kevin Sheng
+author: Kevin Sheng, Ryan Chou
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_fortmoo_platinum_jan16.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_fortmoo_platinum_jan16.html)
 
+## Solution
+
+### Explanation
+
+By computing the [2D Prefix Sum](/silver/more-prefix-sums/) for the grid, we can evaluate the usability of a frame in $\mathcal{O}(1)$ time.
+
+Due to the low bounds on $N$, we can iterate through all possible pairs of columns and find the maximum area frame in $O(N)$ with a sliding window.
+
+### Implementation
+**Time Complexity:** $\mathcal{O}(N^3)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	freopen("fortmoo.in", "r", stdin);
+	freopen("fortmoo.in", "w", stdout);
+
+	int n, m;
+	cin >> n >> m;
+
+	vector<vector<bool>> grid(n, vector<bool>(m, false));
+	vector<vector<int>> cs(n + 1, vector<int>(m + 1, 0));
```

Comment:
wth is `cs`
also remove the 0 and false in the ector declaration

Reply:
cumulative sum

Reply:
rename to pref sim maybe?

Reply:
k